lib/race.rb

```diff
@@ -1,2 +1,40 @@
 class Race
+  attr_reader :office, :candidates, :open
+  
+  def initialize(office)
+    @office = office
+    @candidates = []
+    @open = true
+  end
+
+  def register_candidate!(name_party_hash)
+    candidate = Candidate.new(name_party_hash)
+    @candidates << candidate
+    candidate
+  end
+
+  def open?
+    @open
+  end
+
+  def close!
+    @open = !@open
+  end
+
+  # IDEAS:
+  # I realize i dont have access to the hash formed in Election
+  # because election is not passed into race... race is passed into election.
+
+  # Im thinking i need to form a similar hash in the race class
+  # to then use the enumerable below. Is there an easier way?
```

Comment:
You don't actually need the election object here because each race has its own candidates, and candidates have a `votes` attribute. You're on the right track with max_by but what if you do it on the @candidates attribute instead? Look below for an option:
-
-
-
-
-
```
    return false if open?
    winner = @candidates.max_by do |candidate|
         candidate.votes
    end
```

Reply:
oops, accidental header formatting!